src/app/features/test-amendment-history/test-amendment-history.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, Input } from '@angular/core';
+import { TestResultModel } from '@models/test-result.model';
+
+@Component({
+  selector: 'app-test-amendment-history',
+  templateUrl: './test-amendment-history.component.html',
+  styleUrls: ['./test-amendment-history.component.scss']
+})
+export class TestAmendmentHistoryComponent {
+  @Input() testRecord: TestResultModel | undefined;
+
+  constructor() {}
+
+  getCreatedByName(testResult: TestResultModel | undefined) {
+    return testResult?.createdByName?.length === 0 || !testResult?.createdByName ? testResult?.testerName : testResult?.createdByName;
```

Comment:
Where did this logic come from? Do we know it is correct? What if there is no testerName?

Reply:
I think I saw that logic somewhere, but I can't seem to find it now or may simply have imagined it. 

The ticket links to the [Test_records_fields_to_display spreadsheet](https://dvsa.atlassian.net/wiki/pages/viewpageattachments.action?pageId=18288201). It specifies to show testerName/createdByName. It also says testerName is supposed to be pre-populated with user login details, possibly meaning it is always there?. The [API yml](https://github.com/dvsa/cvs-svc-test-results/blob/develop/docs/test-results-api.yml) description for createdByName states:  `'Applicable only when updating/creating a test from VTM'`. It possibly means it is populated when the form is updated as in [version 3](https://dvsa.atlassian.net/wiki/pages/viewpageattachments.action?pageId=18288201) (now outdated) of the Test_Record_fields_to_display_AS spreadsheet, meaning we want to display that value since it is the last person to edit the record.

I may have just inferred the intended behaviour from the variable names and the information in the spreadsheets and the yml, will raise it in SU to make sure it is right.

Not sure what to do if both createdByName and testerName are null, though